salt/modules/kubernetesmod.py

```diff
@@ -1569,7 +1569,7 @@ def __dict_to_service_spec(spec):
         if key == 'ports':
             spec_obj.ports = []
             for port in value:
-                kube_port = kubernetes.client.V1ServicePort()
+                kube_port = kubernetes.client.V1ServicePort(port=port.get('port', ''))
```

Comment:
This works for setting `port`, but not `node_port` or `target_port` (which map to `nodePort` and `targetPort` in the spec respectively). The constructor accepts a limited enough set of arguments that we could ditch the key iteration logic below and do something like this:

```python
kube_port = kubernetes.client.V1ServicePort(
    name=port.get('name'),
    node_port=port.get('nodePort'),
    port=port.get('port'),
    protocol=port.get('protocol'),
    target_port=port.get('targetPort'),
)
```
It seems like those options are all available in v3.0.0 ([link](https://github.com/kubernetes-client/python/blob/v3.0.0/kubernetes/client/models/v1_service_port.py)).

I found this while testing this exact PR myself with `targetPort` set on a service, so thought it might be worth mentioning. The code above also sets `port` to `None` if it's not defined in the spec, which will result in an error on applying the service by design.